### DISCUSSION TITLE
New Table: Reference Country Classifications

### DISCUSSION THREAD

**Discussion by bill-anderson** [devinit/ddw-analyst-ui#426] [General]
This table allows for cross-referencing of codes, regions and groupings. It contains:
ISO-2 Code,ISO-3 Code,Country Name,Official Name,DAC Recipient Code, FTS Code, UN M49 Code,IMF Code,ISO-Numeric Code,Continent,DAC Recipient Region,WB Region,UN Sub-region,WB Income Group,WB HIPC ,UN LDC,DAC Member,OCED Fragility,Development status,Fuel-exporting,Small island developing State,Landlocked developing,tld,Currency Code,Currency Name,Dial code,Languages,geonameid,neighbours,Capital,Area(sq km),Population 2019,GDP (current US$) 2019,GDP per capita (current US$) 2019
[Reference country classifications.xlsx](https://github.com/devinit/ddw-analyst-ui/files/5210080/Reference.country.classifications.xlsx)

**Comment by bill-anderson** [comment]
I notice that there are various country reference tables. Flagging that these should be removed so that we only maintain a single table. Removal will require checking dependencies on existing datasets first

**Comment by k8hughes** [comment]
@wakibi  and @akmiller01 it would be good for you to both work on this together - or at least to discuss together before the work begins

**Comment by akmiller01** [comment]
Also keen to hear an analyst perspective on a unified country table. What country metadata tables do you tend to use now @Duncan-Knox ? Does the one @bill-anderson posted above fill all of your needs?

**Comment by bill-anderson** [comment]
If it doesn't we should modify it. I'm strongly in favour of a single table for codes and classifications.

NB this table used to contain things like population, GDP, etc. I don't think they belong here, but do we need a table of key country statistics that are used regularly? If not we can always access WB/IMF data as and when we need it.

**Comment by Duncan-Knox** [comment]
Could be one to put to the team more widely. I quite like having the single tables available. This option also allows for building up/storing historical data without needing to add many columns on the single reference table. For example, we have the historical world bank income groups which tracks a income group by country each year since 1987. I also think a single table with the latest information on key country statistics is very useful. Is there an issue with having both file types on the DDW?

**Comment by akmiller01** [comment]
No issue with having similar data in the DDW if they come from different sources. But I would hesitate to put the same data into two different tables. That breaks the "write one, use many times" rule.

**Comment by Duncan-Knox** [comment]
Personally would prefer to keep the individual reference files as well as the unified country table. The individual reference files in some cases can store historical data which would be possibly be more impractical to keep adding to the unified one. I also think the metadata feature is quite useful in providing explanatory and sourcing detail for individual reference files. The search option also allows users to look for a certain country grouping as a dataset. Happy to go with the flow though.

**Comment by bill-anderson** [comment]
I agree we with @akmiller01 that we should always try to store data only once and that a single table should store all current classifications, but @Duncan-Knox has a valid point in storing historical data. Could we still rationalise this to avoid too many random tables floating around?

**Comment by wakibi** [comment]
Hi @Duncan-Knox, what's the structure of the individual reference files? What fields are stored in there? Do they use a similar structure and data types?

**Comment by Duncan-Knox** [comment]
Hi Chris, usually the fields are: country name, ISO2, ISO3, country category, country category 2 (if applicable), and so on.

**Comment by Duncan-Knox** [comment]
This issue was raised at the R&A team meeting and there was support for keeping single country reference files within the DDW as they are, @bill-anderson please can we therefore avoid the deletion of these files, happy to discuss details if needed.
